sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -0,0 +1,438 @@
+                            Settings = settings
+                        };
+
+                    accountInformation._accountName = settingOrDefault(TableConstants.ConnectionStrings.AccountNameSetting);// ?? accountInformation.TableStorageUri.PrimaryUri.Host.Split(new[] { '.' })[0];
+
+                    return true;
+                }
+            }
+
+            // not valid
+            accountInformation = null;
+
+            error("No valid combination of account information found.");
+
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TableConnectionString"/> with development storage credentials using the specified proxy Uri.
+        /// </summary>
+        /// <param name="proxyUri">The proxy endpoint to use.</param>
+        /// <returns>The new <see cref="TableConnectionString"/>.</returns>
+        private static TableConnectionString GetDevelopmentStorageAccount(Uri proxyUri)
+        {
+            UriBuilder builder = proxyUri != null ?
+                new UriBuilder(proxyUri.Scheme, proxyUri.Host) :
+                new UriBuilder("http", TableConstants.ConnectionStrings.Localhost);
+
+            builder.Path = TableConstants.ConnectionStrings.DevStoreAccountName;
+
+            builder.Port = TableConstants.ConnectionStrings.TableEndpointPortNumber;
+            Uri tableEndpoint = builder.Uri;
+
+            builder.Path = TableConstants.ConnectionStrings.DevStoreAccountName + TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix;
+
+            builder.Port = TableConstants.ConnectionStrings.TableEndpointPortNumber;
+            Uri tableSecondaryEndpoint = builder.Uri;
+
+            var credentials = new TableSharedKeyCredential(TableConstants.ConnectionStrings.DevStoreAccountName, TableConstants.ConnectionStrings.DevStoreAccountKey);
+            var account = new TableConnectionString(
+                credentials,
+                tableStorageUri: (tableEndpoint, tableSecondaryEndpoint))
+            {
+                Settings = new Dictionary<string, string>()
+            };
+            account.Settings.Add(TableConstants.ConnectionStrings.UseDevelopmentSetting, "true");
+            if (proxyUri != null)
+            {
+                account.Settings.Add(TableConstants.ConnectionStrings.DevelopmentProxyUriSetting, proxyUri.ToString());
+            }
+
+            account.IsDevStoreAccount = true;
+
+            return account;
+        }
+
+        /// <summary>
+        /// Tokenizes input and stores name value pairs.
+        /// </summary>
+        /// <param name="connectionString">The string to parse.</param>
+        /// <param name="error">Error reporting delegate.</param>
+        /// <returns>Tokenized collection.</returns>
+        private static IDictionary<string, string> ParseStringIntoSettings(string connectionString, Action<string> error)
```

Comment:
We have a shared ConnectionString class for this https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/ConnectionString.cs